### DISCUSSION TITLE
Uploading Large Files

### DISCUSSION THREAD

**Discussion by GunwantBhambra** [rejetto/hfs#269] [Q&A] [answered]
While uploading large files example (15 GB) the uploading fails at around 7% with message file size too large. I have tried to increase and decrease the (min. available disk size) but to no avail.
Is there anything I'm missing here? is there is a limitation to hfs.
version HFS 0.45.0

**Comment by rejetto** [comment]
there's no limitation that I know of with this kind of size.
1. are you on Windows?
2. how much free space on the destination disk?
3. the error message you say is supposed to be given only when the upload starts, not later. If you reach 7% it's my understanding that your upload interrupted, and then you resumed it. Can you confirm?

**Comment by SanokKule** [comment]
Can't confirm on Windows with 0.46.0-beta5
![image](https://github.com/rejetto/hfs/assets/24497821/7226ff49-3252-4091-b7c2-8287fa149b10)

**Comment by GunwantBhambra** [comment, marked as answer]
Solved. Almost
Answering the questions:  Windows, Disk space sufficient. Upload auto-interrupted with the failed message.
The reason found is I am using NGINX reverse proxy for running the website along with HFS on different port. After posting here I gave it a try uploading directly on IP with port opened(DMZ on) and the file is uploaded without error.
So HFS does not have any issues. Thank you for your time.
![image](https://github.com/rejetto/hfs/assets/47209495/77f13b68-a4d6-45bd-bc2b-183470fe385a)

**Comment by Gnbrkm41** [comment]
Actual solution, since I stumbled upon the same issue:
Likely you are using nginx as a reverse proxy, and nginx is limiting the maximum body size.
Solution is to add `client_max_body_size 2G` (adjust as necessary) in the server bock.

**Reply by rejetto** [reply]
yes, this is already included in our guide
https://github.com/rejetto/hfs/wiki/Reverse-proxy
